godel_path_planning/src/trajectory_planning.cpp

```diff
@@ -0,0 +1,196 @@
+#include "godel_path_planning/trajectory_planning.h"
+
+#include <moveit/robot_trajectory/robot_trajectory.h>
+
+#include "descartes_trajectory/cart_trajectory_pt.h"
+#include "descartes_planner/dense_planner.h"
+
+#include <tf_conversions/tf_eigen.h>
+
+#include <descartes_core/robot_model.h>
+#include <pluginlib/class_loader.h>
+
+// Anonymous namespace
+namespace
+{
+  const static double TOOL_POINT_DELAY = 0.75;
+
+  /* This will allow us to adapt to a different robot model by dynamically loading its
+     interface. */
+  static const std::string PLUGIN_NAME = "motoman_sia20d_descartes/MotomanSia20dRobotModel";
+  static pluginlib::ClassLoader<descartes_core::RobotModel> robot_model_loader("descartes_core",
+                                                                     "descartes_core::RobotModel");
+  
+  // Create a descartes RobotModel for graph planning
+  descartes_core::RobotModelPtr createRobotModel(const moveit::core::RobotStatePtr robot_state,
+                                                 const std::string& group_name,
+                                                 const std::string& tool_frame,
+                                                 const std::string& world_frame,
+                                                 const uint8_t iterations)
+  {
+    using descartes_core::RobotModelPtr;
+
+    RobotModelPtr ptr = robot_model_loader.createInstance(PLUGIN_NAME);
+    ptr->initialize("robot_description", group_name, world_frame, tool_frame);
+    return ptr;
+  }
+
+  // Create a CartTrajectoryPt that defines a point & axis with free rotation about z
+  descartes_core::TrajectoryPtPtr pointAxisTrajectoryFactory(double x, double y, double z, 
+                                                             double rx, double ry, double rz,
+                                                             double angle_discretization)
+  {
+    using namespace descartes_core;
+    using namespace descartes_trajectory;
+
+    return boost::shared_ptr<TrajectoryPt>(
+      new CartTrajectoryPt(
+            TolerancedFrame(utils::toFrame(x,y, z, rx, ry, rz, descartes_core::utils::EulerConventions::XYZ),
+             ToleranceBase::zeroTolerance<PositionTolerance>(x, y, z),
+             ToleranceBase::createSymmetric<OrientationTolerance>(rx, ry, 0, 0, 0, 2.0 * M_PI)),
+            0.0, angle_discretization));
+  }
+
+  // Create an axial trajectory pt from a given tf transform
+  descartes_core::TrajectoryPtPtr tfToAxialTrajectoryPt(const tf::Transform& nominal, double discretization)
+  {
+    Eigen::Affine3d eigen_pose;
+    tf::poseTFToEigen(nominal,eigen_pose);
+    Eigen::Vector3d rpy =  eigen_pose.rotation().eulerAngles(0,1,2);
+
+    double rx = rpy(0);
+    double ry = rpy(1);
+    double rz = rpy(2);
+    double x = eigen_pose.translation()(0);
+    double y = eigen_pose.translation()(1);
+    double z = eigen_pose.translation()(2);
+
+    return pointAxisTrajectoryFactory(x, y, z , rx, ry, rz, discretization);
```

Comment:
If the "pointAxisTrajectoryFactory" function isn't ever called outside of here then it should just be part of the calling function.